src/App.jsx

```diff
@@ -1,9 +1,63 @@
-import React from 'react';
+import React, { useState } from 'react';
+
+import List from './List';
+import Input from './Input';
 
 export default function App() {
+  const [state, setState] = useState({
+    id: 0,
+    name: '',
+    category: '',
+    address: '',
+    restaurants: [],
+  });
+
+  const {
+    name,
+    category,
+    address,
+    restaurants,
+  } = state;
+
+  function handleChange(event) {
+    const { name: targetName, value } = event.target;
+    setState({
+      ...state,
+      [targetName]: value,
+    });
+  }
+
+  function handleClick() {
+    setState({
+      id: state.id + 1,
+      name: '',
+      category: '',
+      address: '',
+      restaurants: [
+        ...state.restaurants,
+        {
+          id: state.id,
+          name: state.name,
+          category: state.category,
+          address: state.address,
+        },
```

Comment:
이 부분도 이렇게 처리해줄 수 있을 것 같아요~
```suggestion
        {
          ...state
        },
```

Reply:
엇 그렇네요 리팩토링하다가 놓쳤던 것 같습니다. 감사합니다!

Reply:
고쳐 놓고 알았는데, 이럴 경우 state.restaurants가 restaurants에 다시 네스트되어서 안되겠네요